tests/h/util/session_tracker_test.py

```diff
@@ -0,0 +1,88 @@
+from __future__ import unicode_literals
+
+from uuid import uuid4
+import pytest
+
+from h.db.types import _get_urlsafe_from_hex
+from h.models import Annotation, Document
+from h.util.session_tracker import Tracker, ObjectState
+
+
+def generate_ann_id():
```

Comment:
I'm reusing the same setup and expectations for each test, and this makes it easier to get the IDs in the `expected_changes` fixture.

To recap, at the end of a request, a DB session could be in four possible states:

 - Changes not flushed or committed (indicates a bug)
 - Changes flushed but not committed (indicates a bug)
 - Changes committed
 - Changes rolled back

For each of those four states, there are 3 types of possible change to consider: Added, changed and deleted entries relative to the previous committed state. To avoid having to write 12 separate tests I've reused the same setup and expectations for each.